spec/acceptance/postgresql_psql_spec.rb

```diff
@@ -1,153 +1,143 @@
-    EOS
-
-    apply_manifest(pp, :catch_failures => true)
-    apply_manifest(pp, :catch_changes => true)
-  end
+    MANIFEST
+    it 'does not run SQL when the unless query returns no rows' do
+      apply_manifest(pp_five, catch_failures: true)
+      apply_manifest(pp_five, catch_changes: true)
+    end
 
-  it 'should not run SQL when refreshed and the unless query returns rows' do
-    pp = <<-EOS
+    pp_six = <<-MANIFEST.unindent
       class { 'postgresql::server': } ->
       notify { 'trigger': } ~>
       postgresql_psql { 'foobar':
-        db        => 'postgres',
-        psql_user => 'postgres',
-        command   => 'invalid sql statement',
-        unless    => 'select 1 where 1=1',
+        db          => 'postgres',
+        psql_user   => 'postgres',
+        command     => 'select 1',
+        unless      => 'select 1 where 1=2',
+        refreshonly => true,
       }
-    EOS
-
-    apply_manifest(pp, :catch_failures => true)
-    apply_manifest(pp, :expect_changes => true)
-  end
-
-  context 'with refreshonly' do
-    it 'should not run SQL when the unless query returns no rows' do
-      pp = <<-EOS
-        class { 'postgresql::server': } ->
-        postgresql_psql { 'foobar':
-          db          => 'postgres',
-          psql_user   => 'postgres',
-          command     => 'select 1',
-          unless      => 'select 1 where 1=2',
-          refreshonly => true,
-        }
-      EOS
-
-      apply_manifest(pp, :catch_failures => true)
-      apply_manifest(pp, :catch_changes  => true)
+    MANIFEST
+    it 'runs SQL when refreshed and the unless query returns no rows' do
+      apply_manifest(pp_six, catch_failures: true)
+      apply_manifest(pp_six, expect_changes: true)
     end
 
-    it 'should run SQL when refreshed and the unless query returns no rows' do
-      pp = <<-EOS.unindent
-        class { 'postgresql::server': } ->
-        notify { 'trigger': } ~>
-        postgresql_psql { 'foobar':
-          db          => 'postgres',
-          psql_user   => 'postgres',
-          command     => 'select 1',
-          unless      => 'select 1 where 1=2',
-          refreshonly => true,
-        }
-      EOS
-
-      apply_manifest(pp, :catch_failures => true)
-      apply_manifest(pp, :expect_changes => true)
-    end
-
-    it 'should not run SQL when refreshed and the unless query returns rows' do
-      pp = <<-EOS.unindent
-        class { 'postgresql::server': } ->
-        notify { 'trigger': } ~>
-        postgresql_psql { 'foobar':
-          db          => 'postgres',
-          psql_user   => 'postgres',
-          command     => 'invalid sql query',
-          unless      => 'select 1 where 1=1',
-          refreshonly => true,
-        }
-      EOS
-
-      apply_manifest(pp, :catch_failures => true)
-      apply_manifest(pp, :expect_changes => true)
-    end
-  end
-
-  it 'should not run some SQL when the onlyif query returns no rows' do
-    pp = <<-EOS
+    pp_seven = <<-MANIFEST.unindent
```

Comment:
Do we need to use .unindent?
According to https://zaiste.net/posts/heredoc_in_ruby/  it will 'remove blank characters up to the column where heredoc starts'. I cannot see any blank characters.

Reply:
From what I understand it is to unindent the heredoc to the point were at least one line of it has no indents.

Reply:
That or the first line.

Reply:
This is due to certain heredocs failing when applied if there are excess indents